content/chapters/software-stack/lab/support/app-investigate/get_app_types.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+find /usr/bin /bin /usr/sbin /sbin -type f -exec file {} \; |
+    awk '
+BEGIN {
+    num_bin = 0;
+    num_perl = 0;
+    num_shell = 0;
+    num_python = 0;
+    num_others = 0;
+}
+
+/.*shell script.*/ {
+    num_shell++;
+    next;
+}
+
+/.*Perl script.*/ {
+    num_perl++;
+    next;
+}
+
+/.*ython(|3) script.*/ {
```

Comment:
This regex may not be valid in some regex flavors. Should be:
```suggestion
/.*ython3? script.*/ {
```

Additionally, since it is likely intended to match `python`, it can be improved as:
```suggestion
/.*[Pp]ython3? script.*/ {
```
or, a more complicated alternative for case insensitive searching:
```suggestion
tolower($0) ~ /.*python3? script.*/ {
```